src/runtime/particle-execution-context.ts

```diff
@@ -219,47 +219,94 @@ export class ParticleExecutionContext {
     const p = new Promise<void>(res => resolve = res);
     this.pendingLoads.push(p);
 
-    let particle: Particle;
-    if (spec.implFile && spec.implFile.endsWith('.wasm')) {
-      particle = await this.loadWasmParticle(spec);
-      particle.setCapabilities(this.capabilities(false));
-    } else {
-      const clazz = await this.loader.loadParticleClass(spec);
-      if (!clazz) {
-        return Promise.reject(new Error(`Could not load particle ${id} ${spec.name}`));
-      }
-      particle = new clazz();
-      particle.setCapabilities(this.capabilities(true));
-    }
-    this.particles.push(particle);
+    const particle: Particle = await this.createParticleFromSpec(id, spec);
 
     const handleMap = new Map();
     const registerList: {proxy: StorageProxy, particle: Particle, handle: Handle}[] = [];
 
     proxies.forEach((proxy, name) => {
-      const connSpec = spec.handleConnectionMap.get(name);
-      const handle = handleFor(proxy, this.idGenerator, name, id, connSpec.isInput, connSpec.isOutput);
-      handleMap.set(name, handle);
-
-      // Defer registration of handles with proxies until after particles have a chance to
-      // configure them in setHandles.
-      registerList.push({proxy, particle, handle});
+      this.createHandle(particle, spec, id, name, proxy, handleMap, registerList);
     });
 
     return [particle, async () => {
-      await particle.callSetHandles(handleMap, err => {
-        const exc = new UserException(err, 'setHandles', id, spec.name);
-        this.apiPort.ReportExceptionInHost(exc);
-      });
-      registerList.forEach(({proxy, particle, handle}) => proxy.register(particle, handle));
-      const idx = this.pendingLoads.indexOf(p);
-      this.pendingLoads.splice(idx, 1);
+      await this.assignHandle(particle, spec, id, handleMap, registerList, p);
       resolve();
     }];
   }
 
   private async reloadParticle(id: string) {
-    // TODO(sherrypra): Implement this method.
+    let resolve: Runnable;
+    const p = new Promise<void>(res => resolve = res);
+    this.pendingLoads.push(p);
+
+    // Get the old particle based on the given id and delete the old particle's cache
+    const oldParticle = this.particles.get(id);
+    delete oldParticle.spec.implBlobUrl;
+
+    // Create a new particle and replace the old one
+    const particle: Particle = await this.createParticleFromSpec(id, oldParticle.spec);
+
+    const handleMap = new Map();
+    const registerList: {proxy: StorageProxy, particle: Particle, handle: Handle}[] = [];
+    // Create new handles and disable the handles of the old particles
+    oldParticle.handles.forEach((oldHandle) => {
+      if (oldHandle.storage instanceof StorageProxy) {
```

Comment:
When "oldHandle.storage instanceof StorageProxy" is not the case?

Reply:
Changed the check into right before registering handles (which is the function that store does not have and only storage proxy does). Therefore now new handles will always be created and it will be registered if the instance is a storage proxy :D